homeassistant/components/recorder/__init__.py

```diff
@@ -267,10 +271,13 @@ def do_purge(now):
                     continue
 
             with session_scope(session=self.get_session()) as session:
-                dbevent = Events.from_event(event)
-                session.add(dbevent)
+                dbevent = None
```

Comment:
This is not the correct way to filter out the event. Have a look at the above examples on how things get filtered.